packages/@aws-cdk/aws-codepipeline/test/test.cloudformation-pipeline-actions.ts

```diff
@@ -356,6 +357,53 @@ export = {
   }));
 
   test.done();
+  },
+
+  'Action service role is passed to template'(test: Test) {
+    const stack = new TestFixture();
+
+    const importedRole = Role.import(stack, 'ImportedRole', {
+      roleArn: 'arn:aws:iam::000000000000:role/action-role'
+    });
+    const freshRole = new Role(stack, 'FreshRole', {
+      assumedBy: new ServicePrincipal('magicservice')
+    });
+
+    new PipelineExecuteChangeSetAction(stack.pipeline, 'ImportedRoleAction', {
+      role: importedRole,
+      changeSetName: 'magicSet',
+      stackName: 'magicStack',
+      stage: stack.deployStage
+    });
+
+    new PipelineExecuteChangeSetAction(stack.pipeline, 'FreshRoleAction', {
+      role: freshRole,
+      changeSetName: 'magicSet',
+      stackName: 'magicStack',
+      stage: stack.deployStage
+    });
+
+    expect(stack).to(haveResourceLike('AWS::CodePipeline::Pipeline', {
+      "Stages": [{
+          "Name": "Source" /* don't care about the rest */
+      }, {
+        "Name": "Deploy",
+        "Actions": [{
+          "Name": "ImportedRoleAction",
+          "RoleArn": "arn:aws:iam::000000000000:role/action-role"
+        }, {
+          "Name": "FreshRoleAction",
+          "RoleArn": {
+            "Fn::GetAtt": [
+              "FreshRole472F6E18",
+              "Arn"
+            ]
+          }
+        }]
+      }]
```

Comment:
Can you use standard JS indentation here instead? I have a hard time following what's supposed to be asserted here.

```ts
    expect(stack).to(haveResourceLike('AWS::CodePipeline::Pipeline', {
      "Stages": [
        {
          "Name": "Source" /* don't care about the rest */
        },
        {
          "Name": "Deploy",
          "Actions": [
            {
              "Name": "ImportedRoleAction",
              "RoleArn": "arn:aws:iam::000000000000:role/action-role"
            },
            {
              "Name": "FreshRoleAction",
              "RoleArn": {
                "Fn::GetAtt": [
                  "FreshRole472F6E18",
                  "Arn"
                ],
              }
            }
          ]
        }
      ]
    }));
```